src/libraries/System.Reflection.Emit/tests/PersistableAssemblyBuilder/AssemblySaveILGeneratorTests.cs

```diff
@@ -2174,5 +2174,39 @@ public void RecursiveSumTest()
                 Assert.Equal(55, recursiveMethodFromDisk.Invoke(null, [10]));
             }
         }
+
+        [Fact]
+        public void CallOpenGenericMembersFromConstructedGenericType()
+        {
+            using (TempFile file = TempFile.Create())
+            {
+                AssemblyBuilder ab = AssemblySaveTools.PopulateAssemblyBuilderTypeBuilderAndSaveMethod(out TypeBuilder type, out MethodInfo saveMethod);
+                MethodBuilder method = type.DefineMethod("M1", MethodAttributes.Public, null, null);
+
+                ILGenerator ilGenerator = method.GetILGenerator();
+                LocalBuilder span = ilGenerator.DeclareLocal(typeof(ReadOnlySpan<char>));
+                LocalBuilder str = ilGenerator.DeclareLocal(typeof(string));
+
+                ilGenerator.Emit(OpCodes.Ldstr, "hello");
+                ilGenerator.Emit(OpCodes.Call, typeof(MemoryExtensions).GetMethod("AsSpan", [typeof(string)])!);
+                ilGenerator.Emit(OpCodes.Stloc, span);
+                ilGenerator.Emit(OpCodes.Ldloca_S, span);
+                ilGenerator.Emit(OpCodes.Ldc_I4_1);
+                ilGenerator.Emit(OpCodes.Call, typeof(ReadOnlySpan<char>).GetMethod("Slice", [typeof(int)])!);
+                ilGenerator.Emit(OpCodes.Stloc, span);
+                ilGenerator.Emit(OpCodes.Ldloca_S, span);
+                ilGenerator.Emit(OpCodes.Constrained, typeof(ReadOnlySpan<char>));
+                ilGenerator.Emit(OpCodes.Callvirt, typeof(object).GetMethod("ToString"));
+                ilGenerator.Emit(OpCodes.Stloc, str);
+                ilGenerator.EmitWriteLine(str);
+                ilGenerator.Emit(OpCodes.Ret);
+
+                type.CreateType();
+                saveMethod.Invoke(ab, [file.Path]);
+
+                Type typeFromDisk = Assembly.LoadFile(file.Path).GetType("MyType");
```

Comment:
If we're going to LoadFrom, this test should probably use RemoteExecutor.

Reply:
Is that for getting the file deleted automatically after the test execution or is there other reason?

Reply:
In part. The assembly won't be unloaded, so over time such tests will bloat the process.